### PR TITLE
Docker image guide

### PR DESCRIPTION
**Pull request by sdelamo**
This guide shows the different options to build a docker image with both Gradle and Maven plugins: 

- Docker Image
- AOT Optimized Docker Image
- CRaC Docker Image
- Native Docker Image
- AOT Optimized Native Docker Image

To generate the guide locally: 

`./gradlew micronautDockerImageBuild`

Attach here for convenience as PDF: 

[DockerImageMaven.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10721873/DockerImageMaven.pdf)
[DockerImageGradle.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10721876/DockerImageGradle.pdf)

### CODE REVIEW

File: guides/micronaut-docker-image/micronaut-docker-image.adoc [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+
+:exclude-for-build:gradle
+
+== Building a Docker Image with the Micronaut Maven Plugin
+
+https://micronaut-projects.github.io/micronaut-maven-plugin/latest/[Micronaut Maven Plugin] offers several options to package your application as a Docker Image.
+
+* `docker` builds a Docker image with the application artifacts (compiled classes, resources, dependencies, etc).
+* `docker-native`: builds a Docker image with a GraalVM native image inside.
+* `docker-crac`: build a a pre-warmed, checkpointed Docker Image
```
Comment (alvarosanchez) [inline guides/micronaut-docker-image/micronaut-docker-image.adoc:30]: Perhaps add a note saying that AOT can be enabled via properties?

File: guides/micronaut-docker-image/metadata.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "title": "Building a Docker Image of your Micronaut application",
+  "intro": "Micronaut Build plugins offer several ways to build Docker Images - JAR, Native Executable, CRaC.",
```
Comment (alvarosanchez) [inline guides/micronaut-docker-image/metadata.json]:
```suggestion
  "intro": "Micronaut build plugins offer several ways to build Docker images - JAR, GraalVM native executable, CRaC",
```

File: guides/micronaut-docker-image/metadata.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  "authors": ["Sergio del Amo"],
+  "categories": ["Distribution"],
+  "publicationDate": "2023-02-12",
+  "apps": [
+    {
+      "name": "default",
+      "features": ["graalvm","crac","micronaut-aot"]
```
Comment (alvarosanchez) [inline guides/micronaut-docker-image/metadata.json:11]: I don't see the graalvm feature:

image

Perhaps the PDF is not up-to-date?

File: guides/micronaut-docker-image/micronaut-docker-image.adoc [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+
+:exclude-for-build:
+
+== Sample App
+
+Add a controller which responds with the JSON payload in the apex route.
```
Comment (alvarosanchez) [inline guides/micronaut-docker-image/micronaut-docker-image.adoc]: apex route?

File: guides/micronaut-docker-image/micronaut-docker-image.adoc [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+[source, bash]
+----
+./ttfr.sh micronautguide:latest
+micronautguide:latest
+...
+.
+2.473 seconds
+----
+
+=== Native Docker Image
```
Comment (alvarosanchez) [inline guides/micronaut-docker-image/micronaut-docker-image.adoc]:
```suggestion
=== Docker image with a GraalVM native executable inside
```

File: guides/micronaut-docker-image/micronaut-docker-image.adoc [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+[source, bash]
+----
+./ttfr.sh micronautguide:latest
+micronautguide:latest
+...
+.
+2,359 seconds
+----
+
+=== AOT Optimized Native Docker Image
```
Comment (alvarosanchez) [inline guides/micronaut-docker-image/micronaut-docker-image.adoc]:
```suggestion
=== AOT Optimized Docker Image with a native executable inside
```

**Review by alvarosanchez — APPROVED** [review]

File: guides/micronaut-docker-image/micronaut-docker-image.adoc [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+
+:exclude-for-build:
+
+== Comparisons
+
+Micronaut Framework offers many options for packaging your application as a Docker Image. As illustrated in the following chart, you can speed up your Docker Images by using the https://docs.micronaut.io/latest/guide/#graal[GraalVM integration in the Micronaut Framework], the https://micronaut-projects.github.io/micronaut-crac/latest/guide/[CRaC (Coordinated Restore at checkpoint) integration in  the Micronaut Framework],  https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/#_micronaut_aot_plugin[Micronaut AOT Gradle Plugin] or https://micronaut-projects.github.io/micronaut-maven-plugin/latest/examples/aot.html[Maven Plugin integration with Micronaut AOT].
+
+The following chart illustrates the speed gains you can obtain:
+
+image::docker-image-ttfr.png[]
```
Comment (melix) [inline guides/micronaut-docker-image/micronaut-docker-image.adoc:358]: It is curious that the AOT version is _slower_...

**Review by melix — APPROVED** [review]

File: guides/micronaut-docker-image/micronaut-docker-image.adoc (hunk 4 above)
Comment (sdelamo) [reply guides/micronaut-docker-image/micronaut-docker-image.adoc]: the "/" ? the route route?

File: guides/micronaut-docker-image/micronaut-docker-image.adoc (hunk 7 above)
Comment (sdelamo) [reply guides/micronaut-docker-image/micronaut-docker-image.adoc:358]: AOT is slightly faster in both scenarios. I have [upated the image](https://github.com/micronaut-projects/micronaut-guides/pull/1239/commits/f9a079a6ef1512d0e6a8ed6319e1619efdd26336) and see table

File: guides/micronaut-docker-image/micronaut-docker-image.adoc (hunk 1 above)
Comment (sdelamo) [reply guides/micronaut-docker-image/micronaut-docker-image.adoc:30]: Closed via https://github.com/micronaut-projects/micronaut-guides/pull/1239/commits/4f05ccc83cf78b040653b0985bca8ae207bd6f3a

File: guides/micronaut-docker-image/micronaut-docker-image.adoc (hunk 4 above)
Comment (sdelamo) [reply guides/micronaut-docker-image/micronaut-docker-image.adoc]: https://github.com/micronaut-projects/micronaut-guides/pull/1239/commits/ae1d15bc1c2ffdbcc46a029ea728627acd24f3e3

**Review by alvarosanchez — APPROVED** [review]

**Pull request merged**